server/lib/pubsubNotifiers/pubsubNotifier.js

```diff
@@ -0,0 +1,25 @@
+const notifiers = require('./notifiers')
+
+module.exports = function NewPubSub (pubsubConfig) {
+  const PubSubClass = notifiers[pubsubConfig.type]
+
+  if (!PubSubClass) {
```

Comment:
Is not better encapsulate these if in functions the call them, something like as:
```javascript

module.exports = function NewPubSub (pubsubConfig) {
   const PubSubClass = notifiers[pubsubConfig.type]
  
   try {
          isValidType(PubSubClass);
          isFunction(PubSubClass);
          return getPublish(PubSubClass);
     } catch Error{
          throws Error;
    }

}

```

Reply:
Hi @camilamacedo86 thanks for reviewing! Your suggestion is very good and fair, but I do not see the need to move any of this stuff into separate functions right now because because this is the only place we will ever run these checks.

Reply:
@darahayes the problem is if we don't start the code as it should be then it will not grow as it should be. This code has many ifs, and if someone will improve it will add more ifs then if we encapsulate the logic is not required comments because what it is doing will be clear as we are doing things that allow improvements in a healthy way. Is it something that makes sense?